libcxx/include/__config

```diff
@@ -1460,6 +1461,11 @@ typedef __char32_t char32_t;
 #    define _LIBCPP_HAS_EXPLICIT_THIS_PARAMETER
 #  endif
 
+#  if (defined(__APPLE__) || defined(__FreeBSD__) || defined(__linux__) || defined(_LIBCPP_WIN32API) || defined(_AIX)) &&               \
+      !defined(__PICOLIBC__) && !defined(_LIBCPP_HAS_NO_INCOMPLETE_DEBUGGING)
+#    define _LIBCPP_HAS_DEBUGGING
+#  endif
+
```

Comment:
`is_debugger_present()` is replaceable. I propose to remove `_LIBCPP_HAS_NO_INCOMPLETE_DEBUGGING`/`_LIBCPP_HAS_DEBUGGING` macro and make the function available on any platforms, including platforms that don't have a complete support for it (in our case Picolib only). On these platforms `is_debugger_present()` will return false by default and the user will be able to *replace* it, also if debuggers start supporting `is_debugger_present()`, the function will become usable automatically.